core/domain/stats_domain.py

```diff
@@ -84,3 +88,127 @@ def get_top_answers(self, N):
         return sorted(
             self.answers.iteritems(), key=operator.itemgetter(1),
             reverse=True)[:N]
+
+
+class StateAnswers(object):
+    """Domain object containing answers of states."""
+
+    def __init__(self, exploration_id, exploration_version, state_name,
+                 interaction_id, answers_list):
+        """
+        Initialize domain object for state answers.
+
+        interaction_id contains interaction type of the state, e.g. multiple
+        choice. It is None if no interaction id can be retrieved.
```

Comment:
Yes I think you are right and we should not allow answers if interaction_id is None.